### PR TITLE
libceed 0.7 and occa 1.1.0

### PR DESCRIPTION
**Pull request by jedbrown**

### CODE REVIEW

**Review by dmed256 — APPROVED** [review]
✅ for OCCA

File: var/spack/repos/builtin/packages/libceed/package.py [hunk 1]
```diff
@@ -31,8 +33,10 @@ class Libceed(Package):
     conflicts('+magma', when='@:0.5')
 
     depends_on('cuda', when='+cuda')
+    depends_on('hip@3.8.0', when='@0.7+hip')
```
Comment (v-dobrev) [inline var/spack/repos/builtin/packages/libceed/package.py]: May be better to split this into a conflict:
```python
    conflicts('+hip', when='@:0.6')
```
and allow newer versions of hip with:
```python
    depends_on('hip@3.8.0:', when='+hip')
```

File: var/spack/repos/builtin/packages/libceed/package.py [hunk 2]
```diff
@@ -31,8 +33,10 @@ class Libceed(Package):
...
 
     depends_on('occa@develop', when='@develop+occa')
+    depends_on('occa@1.1.0', when='@0.7:+occa')
```
Comment (v-dobrev) [inline var/spack/repos/builtin/packages/libceed/package.py:41]:
```suggestion
    depends_on('occa@1.1.0:', when='@0.7:+occa')
```

File: var/spack/repos/builtin/packages/libceed/package.py (hunk 1 above)
Comment (jedbrown) [reply var/spack/repos/builtin/packages/libceed/package.py]: That's not very future-proof -- HIP is moving fast and there may well be issues with newer versions. We can add the conflict.

**Review by v-dobrev — APPROVED** [review]
Looks good. Feel to ignore or tweak my suggestions above.

File: var/spack/repos/builtin/packages/libceed/package.py (hunk 2 above)
Comment (jedbrown) [reply var/spack/repos/builtin/packages/libceed/package.py:41]: We've had to track OCCA pretty closely due to interface changes so this seems to invite future breakage. We could do a narrower range, but I'm pretty sure we needed changes when tracking `1.0.*` subminor releases.

**Comment by v-dobrev** [comment]
Another thought: I see that `occa` does not have `+hip` variant yet -- it will be good to add one in the near future.

File: var/spack/repos/builtin/packages/libceed/package.py (hunk 1 above)
Comment (jedbrown) [reply var/spack/repos/builtin/packages/libceed/package.py]: I added the conflict, but not the open-ended HIP version.

Comment (v-dobrev) [reply var/spack/repos/builtin/packages/libceed/package.py]: As it is now, `libceed@develop+hip` will not depend on `hip`. Maybe switch from `'when=@0.7+hip'` to just `'when=+hip'` -- the conflict will take care of the libceed version.

Comment (jedbrown) [reply var/spack/repos/builtin/packages/libceed/package.py]: Seems like more complicated logic to put in version constraints after future releases, so I added this using `depends_on`.

**Pull request merged**